src/index.js

```diff
@@ -4,224 +4,167 @@
 
 'use strict';
 
-// The top-level APIs offered by this library are `ClientConnection` and
-// `ServerConnection` classes.  They each take a callback to be used for
-// sending data to the remote peer, and operate like this:
-//
-//    conn = await ClientConnection.create(psk, pskId, async function send_data_to_server(data) {
-//      // application-specific sending logic here.
-//    })
-//
-//    // Send data to the server by calling `send`,
-//    // which will use the callback provided in the constructor.
-//    // A single `send()` by the application may result in multiple
-//    // invokations of the callback.
-//
-//    await conn.send('application-level data')
-//
-//    // When data is received from the server, push it into
-//    // the connection and let it return any decrypted app-level data.
-//    // There might not be any app-level data if it was protocol control message,
-//    // and the receipt of the data might trigger additional calls to the
-//    // send callback for protocol control purposes.
-//
-//    serverSocket.on('data', async encrypted_data => {
-//      const plaintext = await conn.recv(data)
-//      if (plaintext !== null) {
-//        do_something_with_app_level_data(plaintext)
-//      }
-//    })
-//
-//    // It's good practice to explicitly close the connection
-//    // when finished.  This will send a "closed" notification
-//    // to the server.
-//
-//    await conn.close()
-//
-// The `ServerConnection` API operates similarly; the distinction is mainly
-// in which side is expected to send vs receieve during the protocol handshake.
+// A wrapper that combines a WebSocket to the channelserver
+// with a TLS Connection for encryption.
+
+import {
+  InsecureClientConnection,
+  InsecureServerConnection,
+} from './tlsconnection.js';
 
 import {
-  assert,
-  assertIsBytes,
   bytesToHex,
   hexToBytes,
   bytesToUtf8,
   utf8ToBytes,
-  noop,
 } from './utils.js';
 
-import {
-  getRandomBytes
-} from './crypto.js';
-
-import * as STATE from './states.js';
+// Copy pasta from https://developer.mozilla.org/en-US/docs/Web/API/EventTarget.
+var EventTarget = function() {
+  this.listeners = {};
+};
```

Comment:
I'm kinda sad to find we need to polyfill this, but 👍

Since we're using classes elsewhere here, could this be `class EventTarget` instead, or does it have to be declared via `function`?